src/components/layout/MainNavigation.js

```diff
@@ -0,0 +1,44 @@
+import { Link } from "react-router-dom";
+import { useState } from 'react';
+import NewPostPage from "../../pages/NewPost";
+import classes from "./MainNavigation.module.css";
+import Post from "../UI/Post";
+
+function MainNavigation() {
+    const [showForm, setShowForm] = useState(false);
+  function addForm() {
+      setShowForm(true);
```

Comment:
You might be able to inline these functions, not sure if that would be clearer though as these function names are also acting somewhat like a comment for code clarity

Reply:
Yeah makes sense